core/src/main/java/org/apache/iceberg/view/ViewVersionParser.java

```diff
@@ -0,0 +1,100 @@
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.view;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.JsonNode;
+import java.io.IOException;
+import java.util.Locale;
+import java.util.Map;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableList;
+import org.apache.iceberg.util.JsonUtil;
+
+class ViewVersionParser {
+
+  private static final String VERSION_ID = "version-id";
+  private static final String TIMESTAMP_MS = "timestamp-ms";
+  private static final String SUMMARY = "summary";
+  private static final String OPERATION = "operation";
+
+  private static final String REPRESENTATIONS = "representations";
+
+  private ViewVersionParser() {}
+
+  static void toJson(ViewVersion version, JsonGenerator generator) throws IOException {
+    Preconditions.checkArgument(version != null, "Invalid view version: null");
+    generator.writeStartObject();
+    generator.writeNumberField(VERSION_ID, version.versionId());
+    generator.writeNumberField(TIMESTAMP_MS, version.timestampMillis());
+    JsonUtil.writeStringMap(SUMMARY, version.summary(), generator);
+
+    generator.writeArrayFieldStart(REPRESENTATIONS);
+    for (ViewRepresentation representation : version.representations()) {
+      ViewRepresentationParser.toJson(representation, generator);
+    }
+    generator.writeEndArray();
+
+    generator.writeEndObject();
+  }
+
+  static String toJson(ViewVersion version) {
+    return JsonUtil.generate(gen -> toJson(version, gen), false);
+  }
+
+  static ViewVersion fromJson(String json) {
+    return JsonUtil.parse(json, ViewVersionParser::fromJson);
+  }
+
+  static ViewVersion fromJson(JsonNode node) {
+    Preconditions.checkArgument(node != null, "Cannot parse view version from null object");
+
+    Preconditions.checkArgument(
+        node.isObject(), "Cannot parse table version from a non-object: %s", node);
+
+    int versionId = JsonUtil.getInt(VERSION_ID, node);
+    long timestamp = JsonUtil.getLong(TIMESTAMP_MS, node);
+    Map<String, String> summary = JsonUtil.getStringMap(SUMMARY, node);
+    String operation = summary.get(OPERATION);
+    Preconditions.checkArgument(
+        operation != null, "Cannot parse summary with missing required field: %s", OPERATION);
+
+    try {
+      ViewVersion.Operation.valueOf(operation.toUpperCase(Locale.ENGLISH));
```

Comment:
I need to clean this up, I think operation serializaiton/deserialization should be similar to operations for snapshots. https://github.com/apache/iceberg/blob/master/core/src/main/java/org/apache/iceberg/SnapshotParser.java#L64. since it's required i think it makes sense to directly expose `operation` on the API itself